### ISSUE TITLE
Check for LOS during indicator slam - [merged]

### ISSUE THREAD

**Issue by MaxwellJung** [Overwatch-1-Emulator/ow1-emulator#1561]
In GitLab by @hikiru on Nov 9, 2023, 18:04

_Merges indicator_los_check -> staging_

After some testing, ecksdee's indicator changes keep the player from doing too extreme redirects, instead they usually just get stuck on large objects. I added a rule that drops the player after 0.5 seconds of losing LOS, which should be similar to how OW1 slam worked. I also changed the speed check from 0 to 2, since the player pretty much never reaches 0 and takes a long time to reach 1. But, the speed check should be unnecessary except in very niche cases, since the LOS check should take care of most slams where the player gets stuck.

**Comment by MaxwellJung** [comment]
```suggestion:-0+0
and not eventPlayer.stop_slam) \
```

negation styling more consistent with Python

**Comment by MaxwellJung** [comment]
```suggestion:-0+0
    @Condition not isInLoS(eventPlayer.getEyePosition(), eventPlayer.indicator_slam_position, BarrierLos.PASS_THROUGH_BARRIERS)
```

negation styling more consistent with Python

**Comment by MaxwellJung** [comment]
Is this meant to be commented out?

**Comment by MaxwellJung** [comment]
Looks good other than minor styling changes.

**Comment by MaxwellJung** [comment]
In GitLab by @hikiru on Nov 9, 2023, 18:17

Commented on [src/ow1/heroes/doomfist/slam.opy line 68](https://github.com/Overwatch-1-Emulator/ow1-emulator/compare/adebb6a089276f20754b3498d5ad6f993649b0bb..067ca8cbb8597d25b14be62e89fa2b611ae29a4c#diff-c3ea8c2a418a3dd19cdbba970a815f69R68)



changed this line in [version 2 of the diff](/MaxwellJung/ow1_emulator/-/merge_requests/33/diffs?diff_id=840901893&start_sha=067ca8cbb8597d25b14be62e89fa2b611ae29a4c#7fca664a602119d818c549db0367f308a59f70ed_68_68)

**Comment by MaxwellJung** [comment]
In GitLab by @hikiru on Nov 9, 2023, 18:17

Commented on [src/ow1/heroes/doomfist/slam.opy line 152](https://github.com/Overwatch-1-Emulator/ow1-emulator/compare/adebb6a089276f20754b3498d5ad6f993649b0bb..067ca8cbb8597d25b14be62e89fa2b611ae29a4c#diff-c3ea8c2a418a3dd19cdbba970a815f69R152)



changed this line in [version 2 of the diff](/MaxwellJung/ow1_emulator/-/merge_requests/33/diffs?diff_id=840901893&start_sha=067ca8cbb8597d25b14be62e89fa2b611ae29a4c#7fca664a602119d818c549db0367f308a59f70ed_152_152)

**Comment by MaxwellJung** [comment]
In GitLab by @hikiru on Nov 9, 2023, 18:17

added 1 commit

<ul><li>fa1e940c - styling changes</li></ul>

[Compare with previous version](/MaxwellJung/ow1_emulator/-/merge_requests/33/diffs?diff_id=840901893&start_sha=067ca8cbb8597d25b14be62e89fa2b611ae29a4c)

**Comment by MaxwellJung** [comment]
In GitLab by @hikiru on Nov 9, 2023, 18:17

Commented on [src/ow1/heroes/doomfist/slam.opy line 156](https://github.com/Overwatch-1-Emulator/ow1-emulator/compare/adebb6a089276f20754b3498d5ad6f993649b0bb..fa1e940c4f713abc7ad92b0844816354e7aa962f#diff-c3ea8c2a418a3dd19cdbba970a815f69R156)



oh, nope. i think i forgot to save before i made the branch/merge request, i had it commented out for testing

**Comment by MaxwellJung** [comment]
In GitLab by @hikiru on Nov 9, 2023, 18:20

Commented on [src/ow1/heroes/doomfist/slam.opy line 156](https://github.com/Overwatch-1-Emulator/ow1-emulator/compare/adebb6a089276f20754b3498d5ad6f993649b0bb..fa1e940c4f713abc7ad92b0844816354e7aa962f#diff-c3ea8c2a418a3dd19cdbba970a815f69R156)



changed this line in [version 3 of the diff](/MaxwellJung/ow1_emulator/-/merge_requests/33/diffs?diff_id=840903181&start_sha=fa1e940c4f713abc7ad92b0844816354e7aa962f#7fca664a602119d818c549db0367f308a59f70ed_156_156)

**Comment by MaxwellJung** [comment]
In GitLab by @hikiru on Nov 9, 2023, 18:20

added 2 commits

<ul><li>8dc56909 - Update slam.opy</li><li>18518f43 - Merge branch &#39;indicator_los_check&#39; of...</li></ul>

[Compare with previous version](/MaxwellJung/ow1_emulator/-/merge_requests/33/diffs?diff_id=840903181&start_sha=fa1e940c4f713abc7ad92b0844816354e7aa962f)

**Comment by MaxwellJung** [comment]
resolved all threads

**Comment by MaxwellJung** [comment]
Looks good; thanks for contributing!

**Comment by MaxwellJung** [comment]
approved this merge request

**Comment by MaxwellJung** [comment]
mentioned in commit 869d89ffd65f206353e41c75e6a1a632fa88c297

**Comment by MaxwellJung** [comment]
In GitLab by @ghost1 on Nov 10, 2023, 23:13

:tada: This MR is included in version 1.10.1 :tada:

The release is available on [GitLab release](https://gitlab.com/MaxwellJung/ow1_emulator/-/releases/v1.10.1).

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package: :rocket:
